pyphare/pyphare/pharein/simulation.py

```
@@ -272,6 +272,261 @@ def check_boundaries(ndim, **kwargs):
+        "none",
+        "super-magnetofast-inflow",
+        "super-magnetofast-outflow",
+        "free-pressure-inflow",
+        "fixed-pressure-outflow",
+        "adaptive-outflow",
+    )
+    all_directions = ["x", "y", "z"][:ndim]
+    sides = "lower", "upper"
+    boundary_types = kwargs["boundary_types"]
+    physical_directions = []
+    for dir, type in zip(all_directions, boundary_types):
```

Ruff (0.15.20)

[error] Variable `dir` is shadowing a Python builtin

(A001)